frontend/src/app/main/data/comments.cljs

```diff
@@ -450,6 +451,22 @@
                       (rx/map #(partial fetched-users %))))))
              (rx/catch #(rx/throw {:type :comment-error})))))))
 
+(defn mark-all-threads-as-read
+  "Mark all threads as read"
+  [team-id on-success]
+  (ptk/reify ::mark-all-threads-as-read
+    ev/Event
+    (-data [_] {})
+    ptk/WatchEvent
+    (watch [_ _ _]
+      (let [threads (vals @refs/comment-threads)]
```

Comment:
Refs should not be used on events (at least as general advice). In this case, refs/comment-threads access the `:comment-threads` on the state. A watch event receives state as second argument so you can access the same data directly from state.